src/test/java/org/opensearch/ad/transport/ADJobRunnerTransportActionTests.java

```diff
@@ -0,0 +1,93 @@
+import org.opensearch.jobscheduler.spi.utils.LockService;
+import org.opensearch.jobscheduler.transport.ExtensionJobActionRequest;
+import org.opensearch.jobscheduler.transport.JobRunnerRequest;
+import org.opensearch.sdk.SDKClient.SDKRestClient;
+import org.opensearch.tasks.Task;
+import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.transport.TransportService;
+
+public class ADJobRunnerTransportActionTests extends OpenSearchIntegTestCase {
+
+    private ADJobRunnerTransportAction action;
+
+    private Task task;
+
+    private ActionListener<ExtensionActionResponse> response;
+
+    private ExtensionActionRequest extensionActionRequest;
+
+    private LockService lockService;
+
+    private SDKRestClient sdkRestClient;
+
+    private JobExecutionContext jobExecutionContext;
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+
+        sdkRestClient = mock(SDKRestClient.class);
+        action = new ADJobRunnerTransportAction(mock(TransportService.class), mock(ActionFilters.class), sdkRestClient);
+
+        task = mock(Task.class);
+        lockService = new LockService(mock(Client.class), clusterService());
+        JobDocVersion jobDocVersion = new JobDocVersion(1L, 1L, 1L);
+        Instant time = Instant.ofEpochSecond(1L);
+        jobExecutionContext = new JobExecutionContext(time, jobDocVersion, lockService, "jobIndex", "jobId");
+        JobRunnerRequest jobRunnerRequest = new JobRunnerRequest("token", "jobParameterId", jobExecutionContext);
+        extensionActionRequest = new ExtensionJobActionRequest<>(RestHandlerUtils.EXTENSION_JOB_RUNNER_ACTION_NAME, jobRunnerRequest);
+        response = new ActionListener<>() {
+
+            @Override
+            public void onResponse(ExtensionActionResponse extensionActionResponse) {
+                Assert.assertTrue(true);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                Assert.assertTrue(true);
+            }
+        };
+    }
+
+    @Test
+    public void testJobRunnerTransportAction() {
+        action = new ADJobRunnerTransportAction(mock(TransportService.class), mock(ActionFilters.class), null);
+        action.doExecute(task, extensionActionRequest, response);
+    }
+
+    @Test
+    public void testJobRunnerTransportActionWithNullJobParameterId() throws IOException {
+        JobDocVersion jobDocVersion = new JobDocVersion(1L, 1L, 1L);
+        Instant time = Instant.ofEpochSecond(1L);
+        JobExecutionContext jobExecutionContext = new JobExecutionContext(time, jobDocVersion, lockService, "jobIndex", "jobId");
+        JobRunnerRequest jobRunnerRequest = new JobRunnerRequest("token", "", jobExecutionContext);
+        extensionActionRequest = new ExtensionJobActionRequest<>(RestHandlerUtils.EXTENSION_JOB_RUNNER_ACTION_NAME, jobRunnerRequest);
+
+        action.doExecute(task, extensionActionRequest, response);
```

Comment:
What are you asserting for this test? onResponse and onFailure are hardcoded to `assertTrue(true)`. Since the test is checking the behavior of the ExtensionJobActionRequest with a null jobParameterId, we should ensure that this triggers onFailure

Reply:
response is setup in setUp method. Here we are checking whether it hits onResponse on not.

Reply:
I see that, but the current setup prevents us from differentiating if onResponse or onFailure is triggered, as they are both just going to `assertTrue(true)`. Let's take this particular actionRequest with a null job parameter ID for example, we understand that this request configuration should trigger `onFailure`, yet if it passes, then the test will still assertTrue(true) onResponse and not indicate a failure

Reply:
Same question as @joshpalis here.

Reply:
Okay changes pushed